src/gpu/intel/jit/v2/conv/kernel_desc_2d_reqs.cpp

```diff
@@ -169,12 +184,12 @@ void generate_2d_reqs(const kernel_desc_t &desc, tensor_kind_t tensor_kind,
         reqs.add_no_simplify(H % params.y_stride == 0);
         reqs.add_no_simplify(H / params.y_stride <= block_2d_max_dim());
     }
-    reqs.add_no_simplify(P >= safe_div(block_2d_min_dim(), type_size));
-    reqs.add_no_simplify(P <= safe_div(block_2d_max_dim(), type_size));
+    reqs.add_no_simplify(P.ge(safe_div(block_2d_min_dim(), type_size)));
+    reqs.add_no_simplify(P.le(safe_div(block_2d_max_dim(), type_size)));
     reqs.add_no_simplify(
-            P % safe_div(block_2d_pitch_alignment(desc.hw_desc), type_size)
+            P.mod(safe_div(block_2d_pitch_alignment(desc.hw_desc), type_size))
```

Comment:
nit: Adding all these new binary functions seems like overkill compared to just adding a new variable (`P_expr`?) with the appropriate type.

Reply:
I misunderstood exactly what was happing here. Thanks for the explanation.